lib/rucio/core/rule.py

```diff
@@ -1285,14 +1286,24 @@ def update_rule(rule_id, options, session=None):
 
             elif key == 'child_rule_id':
                 # Check if the child rule has the same scope/name as the parent rule
-                child_rule = session.query(models.ReplicationRule).filter_by(id=options[key]).one()
-                if rule.scope != child_rule.scope or rule.name != child_rule.name:
-                    raise InputValidationError('Parent and child rule must be set on the same dataset.')
-
-                if rule.id == options[key]:
-                    raise InputValidationError('Self-referencing parent/child-relationship.')
-                if child_rule.state != RuleState.OK:
-                    rule.child_rule_id = options[key]
+                child_id: Optional[str] = options[key]
+                if child_id is None:
+                    # dissolve exisiting relationship
+                    if not rule.child_rule_id:
+                        raise InputValidationError('Cannot detach child when no such relationship exists')
+                    # dissolve relationship
+                    rule.child_rule_id = None  # type: ignore
+                    # remove expiration date
+                    rule.expires_at = None  # type: ignore
+                else:
+                    child_rule = (session.query(models.ReplicationRule).
+                                  filter_by(id=child_id).one())
```

Comment:
Bonus points if you use the SQLAlchemy 2.0 syntax instead! Refer to #5247 if you’re willing, but note that it would probably have to go into its own commit.